[libc] Stop duplicating wcschr().

enh-google opened the pull request:
Three implementations of wcschr() is two too many.

enh-google committed:
[libc] Stop duplicating wcschr().

lntue commented on libc/src/wchar/wcschr.cpp:
```diff
@@ -11,15 +11,12 @@
-  if (*s == c)
-    return s;
-  return nullptr;
+  return internal::wcschr(s, c);
```
since you are here, can you add `LIBC_CRASH_ON_NULLPTR(s);` and its header+dependency?

enh-google committed:
[libc] add missing includes and LIBC_CRASH_ON_NULLPTR().

lntue commented:
errors from the pre-commit:
```
In file included from /home/runner/work/llvm-project/llvm-project/libc/src/wchar/wcstok.cpp:13:
/home/runner/work/llvm-project/llvm-project/libc/src/wchar/wchar_utils.h:23:10: error: cannot initialize return object of type 'wchar_t *' with an rvalue of type 'const wchar_t *'
   23 |   return (*s == c) ? s : nullptr;
      |          ^~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/llvm-project/llvm-project/libc/src/wchar/wchar_utils.h:32:19: error: no matching function for call to 'wcschr'
   32 |     bool in_set = wcschr(s1[i], s2);
      |                   ^~~~~~
/home/runner/work/llvm-project/llvm-project/libc/src/wchar/wchar_utils.h:20:29: note: candidate function not viable: no known conversion from 'const wchar_t' to 'const wchar_t *' for 1st argument; take the address of the argument with &
   20 | LIBC_INLINE static wchar_t *wcschr(const wchar_t *s, wchar_t c) {
      |                             ^      ~~~~~~~~~~~~~~~~
   21 |   for (; *s && *s != c; ++s)
   22 |     ;
   23 |   return (*s == c) ? s : nullptr;
   24 | }
   ```

enh-google committed:
[libc] Fix internal::wcschr() return type.

The exported function is using a similar white lie for convenience, so the internal function may as well rather than adding const casts.

uzairnawaz commented on libc/src/wchar/wcstok.cpp:
```diff
@@ -31,11 +25,11 @@ LLVM_LIBC_FUNCTION(wchar_t *, wcstok,
 
   wchar_t *tok_start, *tok_end;
-  for (tok_start = str; *tok_start != L'\0' && isADelimeter(*tok_start, delim);
+  for (tok_start = str; *tok_start != L'\0' && wcschr(delims, *tok_start);
```
Should this be `internal::wcschr`? (here and below)

uzairnawaz reviewed (commented):
Thanks for the fixes, I didn't realize how often we were rewriting the same code 😅

enh-google committed:
[libc] wcstok(): explicitly say internal::wcschr().

enh-google committed:
[libc] fix wcstok() formatting.

frobtech reviewed (approved):
lgtm. We need a clang-tidy checker to reject `static` used in headers.
